app/src/main/java/se/hellsoft/android/instantsearchdemo/SearchViewModel.kt

```diff
@@ -70,6 +66,9 @@ class SearchViewModel(
     @ExperimentalCoroutinesApi
     val searchResult = internalSearchResult.asLiveData()
 
+    @ExperimentalCoroutinesApi
+    suspend fun onSearchTextChanged(text: String) = queryChannel.send(text)
```

Comment:
💭 I actually really like this. Now the implementation detail (channel) is really a detail and the view doesn't get exposed to the entire channel API as before.